dbt/include/spark/macros/adapters.sql

```diff
@@ -160,6 +163,53 @@
 {%- endmacro -%}
 
 
+{% macro persist_constraints(relation, model) %}
+  {{ return(adapter.dispatch('persist_constraints', 'dbt')(relation, model)) }}
+{% endmacro %}
+
+{% macro spark__persist_constraints(relation, model) %}
+  {% if config.get('constraints_enabled', False) and config.get('file_format', 'delta') == 'delta' %}
+    {% do alter_table_add_constraints(relation, model.columns) %}
+    {% do alter_column_set_constraints(relation, model.columns) %}
+  {% endif %}
+{% endmacro %}
+
+{% macro alter_table_add_constraints(relation, constraints) %}
+  {{ return(adapter.dispatch('alter_table_add_constraints', 'dbt')(relation, constraints)) }}
+{% endmacro %}
+
+{% macro spark__alter_table_add_constraints(relation, column_dict) %}
+
+  {% for column_name in column_dict %}
+    {% set constraints_check = column_dict[column_name]['constraints_check'] %}
+    {% for constraint_check in constraints_check %}
+      {%- set constraint_hash = local_md5(column_name ~ ";" ~ constraint_check) -%}
+      {% if not is_incremental() %}
+        {% call statement() %}
+          alter table {{ relation }} add constraint {{ constraint_hash }} check ({{ column_name }} {{ constraint_check }});
```

Comment:
Should users define the check including the column name, or not? In [the current implementation](https://github.com/dbt-labs/dbt-core/blob/6806a7c86edf68f87ae3563606da3a5990758af1/tests/adapter/dbt/tests/adapter/constraints/fixtures.py#L52), it is included, so it wold be repeated here

Reply:
I am first trying to get the tests passing with it included but in my mind it doesn't make sense to add the column name. We technically today can put the `check` of a given column under another one which seems odd to me.

Reply:
Even though you can add a `check` to another column, we shouldn't limit developers. The `check` inline with the respective column provides reasonable signals that there should be a 1:1 mapping even if we don't enforce it.